framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -483,23 +526,85 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
             }
         }
 
-        // Check type-system-specific annotations.
+        // Check for type-system-specific annotations.
         List<Pair<AnnotationMirror, AnnotationMirror>> declAnnotations =
                 factory.getDeclAnnotationWithMetaAnnotation(
                         methodElement, ConditionalPostconditionAnnotation.class);
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
-            if (postcondAnno == null) {
-                continue;
-            }
-            List<String> expressions =
-                    AnnotationUtils.getElementValueArray(anno, "expression", String.class, false);
-            boolean annoResult =
-                    AnnotationUtils.getElementValue(anno, "result", Boolean.class, false);
-            for (String expr : expressions) {
-                result.add(new ConditionalPostcondition(expr, annoResult, postcondAnno, anno));
+
+            result.addAll(getConditionalPostConditionAnnotations(anno, metaAnno));
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns a collection of conditional postcondition annotations that can consist of
```

Comment:
Similar comment as above about the Javadoc comment.

Also a similar comment about the logic:  How about doing a single test outside the `for` loop rather than a test on each iteration through the `for` loop?

Reply:
Removed for loop.